classes/entity/enemy/koopa/parakoopa.gd

```diff
@@ -82,26 +78,26 @@ func _on_Damage_body_entered(body):
 
 
 func spawn_shell(body):
-	$Kick.play()
+	change_state()
 	body.vel.y = -5
 	get_parent().call_deferred("add_child", shell)
 	shell.position = position + Vector2(0, 7.5)
-	shell.color = color
 	if body.global_position.x < global_position.x:
 		shell.vel.x = 5
 	else:
 		shell.vel.x = -5
-	$TopCollision.set_deferred("monitoring", false)
-	$Damage.set_deferred("monitoring", false)
-	set_deferred("visible", false)
+
+
+# Called whenever the parakoopa loses its wings
+func change_state():
```

Comment:
Would suggest calling this `defeat()` instead. `change_state()` implies that there's more than one state it could go to.